testgrid/tgrun/pkg/runner/metrics.go

```diff
@@ -0,0 +1,91 @@
+package runner
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/pkg/errors"
+	tghandlers "github.com/replicatedhq/kurl/testgrid/tgapi/pkg/handlers"
+	"github.com/replicatedhq/kurl/testgrid/tgrun/pkg/runner/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+)
+
+func ReportMetrics(runnerOptions types.RunnerOptions) error {
+	runningVMIs, err := countRunningVMIs()
+	if err != nil {
+		return errors.Wrap(err, "failed to count running vmis")
+	}
+
+	freeCPU, freeRAM, err := getFreeResources()
+	if err != nil {
+		return errors.Wrap(err, "failed to get free resources")
+	}
+
+	hostName, err := os.Hostname()
+	if err != nil {
+		return errors.Wrap(err, "failed to get hostname")
+	}
+
+	runnerStatusRequest := tghandlers.RunnerStatusRequest{
+		FreeCPU:      freeCPU,
+		FreeRAM:      freeRAM,
+		RunningTests: float64(runningVMIs),
+		Hostname:     hostName,
+	}
+	b, err := json.Marshal(runnerStatusRequest)
```

Comment:
handle error here?

Reply:
ah yeah, we should
not that there's much chance of an error with that object though (no chance of circular dependencies)